source/runner/Dangerfile.ts

```diff
@@ -54,6 +54,18 @@ export interface DangerContext {
    * @param {number | undefined} line the line which this message should be attached to
    */
   message(message: MarkdownString, file?: string, line?: number): void
+  /**
+   * Adds a message to the Danger table, the only difference between this
+   * and warn is the default emoji which shows in the table.
+   * You can also specifiy a custom emoji to show in the table for each message
+   * 
+   * @param {MarkdownString} message the String to output
+   * @param {{file?: string, line?: string, icon?: MarkdownString}} [opts]
+   * @param opts.file a file which this message should be attached to
+   * @param opts.line the line which this message should be attached to
+   * @param opts.icon icon string or image to show in table, take care not to break table formatting
+   */
+  message(message: MarkdownString, opts?: { file?: string, line?: number, icon?: MarkdownString }): void
```

Comment:
Interesting, yeah, I can see this as a reasonable approach, it is getting pretty unwieldy at that point 👍